content/techniques/mongo.md

```diff
@@ -173,6 +171,94 @@ export class CatsService {
 }
 ```
 
+#### Hooks (middleware)
+
+Middleware (also called pre and post hooks) are functions which are passed control during execution of asynchronous functions. Middleware is specified on the schema level and is useful for writing plugins ([source](https://mongoosejs.com/docs/middleware.html)). Calling `pre()` or `post()` after compiling a model does not work in Mongoose. In order to register a hook **before** model registration, use the `forFeatureAsync()` method.
```

Comment:
Suggestion: to be a little more explicit, please consider this change:
To register a hook **before** model registration, use the `forFeatureAsync()` method of the `MongooseModule` along with a factory provider (i.e., `useFactory`).  With this technique, you can access a `schema` object, then use the `pre()` or `post()` method to register a hook on that schema.  See example below: